### PR TITLE
[Draft under review] New master

### PR DESCRIPTION
**Pull request by Tirtha-Samsung**

### CODE REVIEW

**Comment by sunghan-chang** [comment]
@Tirtha-Samsung Those commit titles are not allowed in TizenRT. We can't know what they are. Please write them to know what they are briefly.

**Comment by sunghan-chang** [comment]
@Tirtha-Samsung How about PR #5671 ? Do you make this instead of that? If yes, let's close it.

**Comment by sunghan-chang** [comment]
@Tirtha-Samsung And this PR has wrong modifications like change in logdump.c. Need to check once more

File: os/arch/arm/src/amebad/Make.defs [hunk 1]
```diff
@@ -200,6 +200,10 @@ EXTRA_LIBS += ${TOPDIR}/board/rtl8721csm/src/libs/cmse_implib.a
 endif
 endif
 
+ifeq ($(CONFIG_RTK_WLAN),y)
+EXTRA_LIBS += $(TOPDIR)/drivers/wireless/realtek/librtl.a
+endif
```
Comment (sunghan-chang) [inline os/arch/arm/src/amebad/Make.defs:205]: This should be reverted.

File: os/board/rtl8721csm/src/component/common/bluetooth/realtek/sdk/example/ble_scatternet/ble_tizenrt_scatternet_app.c [hunk 2]
```diff
@@ -1687,6 +1687,7 @@ T_APP_RESULT ble_tizenrt_scatternet_gcs_client_callback(T_CLIENT_ID client_id, u
                         }
                     } else {
                         debug_print("Memory allocation failed \n");
+                        os_mem_free(notify_result);
```
Comment (sunghan-chang) [inline os/board/rtl8721csm/src/component/common/bluetooth/realtek/sdk/example/ble_scatternet/ble_tizenrt_scatternet_app.c:1690]: Could you split this as different commit? Because this is the change on RTL8721CSM, not RTL8720E.

File: build/tools/amebalite/gnu_utility/AP2_SWDPWD.JLinkScript [hunk 3]
```diff
@@ -0,0 +1,240 @@
+/*********************************************************************
```
Comment (sunghan-chang) [inline build/tools/amebalite/gnu_utility/AP2_SWDPWD.JLinkScript:1]: Let's move the change in build into 4th commit (change in build folder including build/configs)

File: external/webserver/http_client.c [hunk 4]
```diff
@@ -229,7 +229,7 @@ static void parse_keep_alive_header(struct http_client_t *client, struct http_ke
 	char *header_field_value = NULL;
 
 	if (client->keep_alive_header_flag == 1) {
-		HTTP_LOGD("keep alive header is already set: timeout[%d] max[%d] \n",
+		HTTP_LOGE("keep alive header is already set: timeout[%d] max[%d] \n",
```
Comment (sunghan-chang) [inline external/webserver/http_client.c:232]: Let's revert this file.

File: os/FlatLibs.mk [hunk 5]
```diff
@@ -161,6 +161,37 @@ TINYARALIBS += $(LIBRARIES_DIR)$(DELIM)libtuv$(LIBEXT)
 TINYARALIBS += $(LIBRARIES_DIR)$(DELIM)libjerry-libm$(LIBEXT)
 endif
 
+# Add library for external bcm support.
```
Comment (sunghan-chang) [inline os/FlatLibs.mk:164]: Let's revert this file.

File: os/kernel/log_dump/log_dump.c [hunk 6]
```diff
@@ -477,7 +477,6 @@ int log_dump(int argc, char *argv[])
 			compress_last_block = false;
 		} else {
 			/* compress the completely filled block */
-			writesize = CONFIG_LOG_DUMP_CHUNK_SIZE;
```
Comment (sunghan-chang) [inline os/kernel/log_dump/log_dump.c:480]: Let's revert this file

**Comment by sunghan-chang** [comment]
@Tirtha-Samsung Noone has interesting on origin branch `RTL8720E_Support ->`. Let's say `os/arch: RTL8720E chip architecture support`, `os/board: RTL8720E BSP support` and so on. Here is the public github. Let's leave meaningful title for user.

**Comment by sunghan-chang** [comment]
Anf there are still build breaks. I think you don't modify what I left. Please check them and resolve build breaks.

File: os/arch/arm/src/amebad/Make.defs (hunk 1 above)
Comment (amogh-samsung) [reply os/arch/arm/src/amebad/Make.defs]: Done

File: os/board/rtl8721csm/src/component/common/bluetooth/realtek/sdk/example/ble_scatternet/ble_tizenrt_scatternet_app.c (hunk 2 above)
Comment (amogh-samsung) [reply os/board/rtl8721csm/src/component/common/bluetooth/realtek/sdk/example/ble_scatternet/ble_tizenrt_scatternet_app.c]: Left out for now

File: external/webserver/http_client.c (hunk 4 above)
Comment (amogh-samsung) [reply external/webserver/http_client.c]: Done

File: os/FlatLibs.mk (hunk 5 above)
Comment (amogh-samsung) [reply os/FlatLibs.mk]: Done

**Comment by amogh-samsung** [comment]
> @Tirtha-Samsung Noone has interesting on origin branch `RTL8720E_Support ->`. Let's say `os/arch: RTL8720E chip architecture support`, `os/board: RTL8720E BSP support` and so on. Here is the public github. Let's leave meaningful title for user.

Taken care of, commits reworded and added numbers to indicate that all must be applied together for proper support.

File: os/include/tinyara/wifi/rtk/wifi_constants.h [hunk 7]
```diff
@@ -56,6 +56,10 @@ extern "C" {
 
 #define MCSSET_LEN			16
 
+
```
Comment (sunghan-chang) [inline os/include/tinyara/wifi/rtk/wifi_constants.h:59]: Let's add this file in bsp change.

File: os/dbuild.sh [hunk 8]
```diff
@@ -84,6 +84,8 @@ function FIND_BINFILE()
 			BINFILE="${BINDIR}/tinyara_master_strip"
 		elif [[ "${CONFIG_ARCH_BOARD}" == "rtl8721csm" ]]; then
 			BINFILE="${BINDIR}/${KERNEL_BIN_NAME}"
+		elif [[ "${CONFIG_ARCH_BOARD}" == "rtl8720e" ]]; then
```
Comment (sunghan-chang) [inline os/dbuild.sh:87]: how about adding this in build change?

File: lib/libc/machine/arm/Make.defs [hunk 9]
```diff
@@ -60,6 +60,6 @@ else ifeq ($(CONFIG_ARCH_CORTEXM4),y)   # Cortex-M4 is ARMv7E-M
 include ${TOPDIR}/../lib/libc/machine/arm/armv7-m/Make.defs
 else ifeq ($(CONFIG_ARCH_CORTEXM7),y)   # Cortex-M7 is ARMv7E-M
 include ${TOPDIR}/../lib/libc/machine/arm/armv7-m/Make.defs
-else ifeq ($(CONFIG_ARCH_CORTEXM33),y)   # Cortex-M33 is ARMv7E-M
+else ifeq ($(CONFIG_ARCH_ARMV8M_FAMILY),y)   # Cortex-M33/M55 is ARMv8-M
```
Comment (sunghan-chang) [inline lib/libc/machine/arm/Make.defs:63]: Let's add this in arch change

**Comment by sunghan-chang** [comment]
@amogh-samsung Thank you for update. Except above, no more comment.

**Comment by abhishek-samsung** [comment]
Hi @amogh-samsung  and @Tirtha-Samsung, could you please modify the PR title too? **New master** --> **Add support for RTL8720E**

File: os/include/tinyara/wifi/rtk/wifi_constants.h (hunk 7 above)
Comment (amogh-samsung) [reply os/include/tinyara/wifi/rtk/wifi_constants.h:59]: You want it to be separated from this PR?

File: os/dbuild.sh (hunk 8 above)
Comment (amogh-samsung) [reply os/dbuild.sh:87]: Done

File: lib/libc/machine/arm/Make.defs (hunk 9 above)
Comment (amogh-samsung) [reply lib/libc/machine/arm/Make.defs:63]: Lets do this for all 3 lib/libc/* files in the apps commit
Moving all 3 to the arch support commit.

File: os/include/tinyara/wifi/rtk/wifi_constants.h (hunk 7 above)
Comment (sunghan-chang) [reply os/include/tinyara/wifi/rtk/wifi_constants.h:59]: No. Let's add it in `[2/5]os/board: Add support for Rtl8720e board`.
